src/core/error_handling.ts

```diff
@@ -127,6 +127,42 @@ API.fatal_error = function (e: any) {
   throw e;
 };
 
+class FatalPyodideError extends Error {}
+Object.defineProperty(FatalPyodideError.prototype, "name", {
+  value: FatalPyodideError.name,
+});
+
+let stderr_chars: number[] = [];
+API.capture_stderr = function () {
+  stderr_chars = [];
+  const FS = Module.FS;
+  FS.createDevice("/dev", "capture_stderr", null, (e: number) =>
+    stderr_chars.push(e)
+  );
+  FS.closeStream(2);
+  FS.open("/dev/capture_stderr", 1);
```

Comment:
Interesting. So we close stderr fd (2), then open another device so it gets the just closed stderr fd...

Reply:
Yes. As far as I can tell there is no obvious better way to do this (`open` can't specify a file descriptor, `dup` isn't available from the JavaScript API and `FS.createStream` is a bit low level). Probably deserves a comment.